sympy/stats/stochastic_process_types.py

```diff
@@ -169,9 +171,21 @@ def state_space(self) -> tUnion[FiniteSet, Range]:
             return FiniteSet(*self.args[1])
         return self.args[1]
 
-    @property
-    def distribution(self):
-        return None
+    def _deprecation_warn_distribution(self):
+        SymPyDeprecationWarning(
+            feature="Calling distribution with RandomIndexedSymbol",
+            useinstead="distribution with just Symbol as argument",
```

Comment:
```suggestion
            useinstead="distribution with just timestamp as argument",
```
It may not be necessary that timestamp is just a `Symbol` object, it may be an integer, float too.